### ISSUE TITLE
Implementation in BaseActivity

### ISSUE THREAD

**Issue by ryderz8** [motorro/AppUpdateWrapper#13]
Hi, Thanks for the wonderful work.

I have to implement this in the Multi-activity application. So, in BaseActivity how is that **withUpdatesCheck** boolean working(when to set false/true). Also, since the **checkForUpdate()** method in BaseActivity we are forcing for FLEXIBLE update everytime, so can we instead check which type of update is available either IMMEDIATE or FLEXIBLE and accordingly proceed?

It will be really helpful if you could provide some insights on how to work with the Multiple activities approach.

Thanks

**Comment by motorro** [comment]
@ryderz8 

## Update type

### TLDR
It's upon yourself which one to choose
* Use IMMEDIATE when your application version is not supported anymore. Say, you check some back-end method and it tells you your app is outdated
* Use FLEXIBLE every time you open an application - just to inform user that there is an update. Update is not mandatory this way.

### Official info
AFAIK the update flow Flexible/Immediate is some kind of unfinished and undocumented concept by google. Although the `AppUpdateManager` returns particular update flow availability through update check - I found no way to set the update type when you roll-out an update to google play. That was in times the library was created. Later-on google updated their [documentation](https://developer.android.com/guide/playcore/in-app-updates) stating that Immediate update is the one user should perform at once to continue using the app. Which confirmed my initial assumption. 
So it is actually at your choice which one to use - google does not tell you this (at least for a moment).
I use the following way to choose:
- Open app
- Go to my backend reporting current app version
- Backend answers 
-- STOP. NOT SUPPORTED - go IMMEDIATE
-- OK - go FLEXIBLE

## Multiple activities
Refer to [this section](https://github.com/motorro/AppUpdateWrapper#using-library-in-multi-activity-setup) for more information. Basically, you just start flexible flow in every activity you want to support updates in. The wrapper will catch-up with the play-core update state if it is already running. This `withUpdateCheck` flag in the gist you mention from a base activity is just a short-cut to starting flexible flow.
